### PR TITLE
Write in update-markdown command for registry

### PR DESCRIPTION
**Pull request by jsuereth**
After this PR, we can start trying out the update-markdown command against semconv registry.

- Create new helper method to deal with possible git repo location.
- Adds file-walking all markdown files
- For now, panics on failure to ensure bad exit from process.
- Adds dry_run argument for validation vs. actual update.

### CODE REVIEW

File: src/registry/update_markdown.rs [hunk 1]
```diff
@@ -25,13 +26,53 @@ pub struct RegistryUpdateMarkdownArgs {
...
+    let registry = ResolvedSemconvRegistry::try_from_url(
+        args.registry.clone(),
+        args.registry_git_sub_dir.clone(),
+        cache,
+        log.clone(),
+    )
+    .unwrap_or_else(|e| {
+        panic!("Failed to resolve the semantic convention registry.\n{e}");
+    });
+    log.success("Registry resolved succesfully");
```
Comment (lquerel) [inline src/registry/update_markdown.rs:50]:
```suggestion
    log.success("Registry resolved successfully");
```

File: src/registry/update_markdown.rs [hunk 2]
```diff
@@ -25,13 +26,53 @@ pub struct RegistryUpdateMarkdownArgs {
...
+    {
+        log.info(&format!("{}: ${}", operation, entry.path().display()));
+        if let Err(error) =
+            update_markdown(&entry.path().display().to_string(), &registry, args.dry_run)
+        {
+            has_error = true;
+            log.error(&format!("{error}"));
+        }
+    }
+    panic!("weaver registry update-markdown failed.")
```
Comment (lquerel) [inline src/registry/update_markdown.rs:77]: I don't understand this panic! at the end of the command.

**Review by lquerel — COMMENTED** [review]
I don't understand how this command can end without a panic. I might miss something obvious, but I don't see it :-)

File: src/registry/update_markdown.rs (hunk 2 above)
Comment (jsuereth) [reply src/registry/update_markdown.rs]: Fixed

**Review by lquerel — APPROVED** [review]

File: src/registry/update_markdown.rs (hunk 2 above)
Comment (lquerel) [reply src/registry/update_markdown.rs]: Yes either panic! or process::exit(1). (see https://doc.rust-lang.org/std/process/fn.exit.html)

**Pull request merged**